contrib/codegen/templates/output/output_snapshot.gotmpl

```diff
@@ -453,6 +453,11 @@ type Builder interface {
     // this can be used to collect clusters for use with MultiCluster snapshots.
     AddCluster(cluster string)
 
+    // merge all the resources from anotehr Builder into this one
```

Comment:
```suggestion
    // merge all the resources from another Builder into this one
```

Also, can you possibly put an explanation as to how the merge occurs, which resources take precedence, etc.